File: app/src/main/java/com/example/cbanda/achieve/presentation/list/GoalListViewModel.java
package com.example.cbanda.achieve.presentation.list;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.example.cbanda.achieve.models.db.Goal;
import com.example.cbanda.achieve.models.repository.GoalRepository;
import com.example.cbanda.achieve.utilities.InjectorUtils;

import java.util.List;

public class GoalListViewModel extends AndroidViewModel {

    private GoalRepository goalRepository;

    public GoalListViewModel(@NonNull Application application) {
        super(application);
        goalRepository = InjectorUtils.getInstance(application);
    }

    LiveData<List<Goal>> getGoals() {
        return goalRepository.getListGoals();
    }

}

File: app/src/main/java/com/example/cbanda/achieve/presentation/list/GoalListViewModel.kt
package com.example.cbanda.achieve.presentation.list

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.example.cbanda.achieve.models.db.Goal
import com.example.cbanda.achieve.models.repository.GoalRepository
import com.example.cbanda.achieve.utilities.InjectorUtils

class GoalListViewModel(application: Application) : AndroidViewModel(application) {



    private val goalRepository: GoalRepository = InjectorUtils.getInstance(application)

    private val goals: LiveData<List<Goal>> = goalRepository.listGoals

    fun getGoals(): LiveData<List<Goal>> {
        return goalRepository.listGoals
    }


}
